src/components/Card/favoritesCard.js

```diff
@@ -13,90 +13,80 @@ const IndexCalendar = dynamic(() => import('../Calendar/indexCalendar'), {
 import StyledCardHeader from '../CardHeader/styled';
 import StyledCardHeadline from '../CardHeadline/styled';
 
 import StyledCard from './styled';
 
-export default function FavoritesCard({card}) {
-	const changeSets = useStore(state => state.changeSets);
-	const changeReps = useStore(state => state.changeReps);
-	const changeWeight = useStore(state => state.changeWeight);
-	const debookmark = useStore(state => state.debookmark);
-	const [isShown, setIsShown] = useState(false);
-	const [showText, setShowText] = useState(false);
-	let [buttonText, setButtonText] = useState(true);
-	const [isBookmarked, setIsBookmarked] = useState(true);
-
-	const handleChange = () => {
-		return setButtonText(!buttonText);
-	};
-
-	function handleSubmit(event) {
-		event.preventDefault();
-	}
-	function handleSetInput(input) {
-		changeSets(card.id, Number(input));
-	}
-
-	function handleRepInput(input) {
-		changeReps(card.id, Number(input));
-	}
+export default function FavoritesCard({exerciseCard}) {
+	const handleBookmark = useStore(state => state.handleBookmark);
+	const [isCalendarVisible, setIsCalendarVisible] = useState(false);
+	const [isInstructionVisible, setIsInstructionVisible] = useState(false);
+	const addToPlanner = useStore(state => state.addToPlanner);
+	const workouts = useStore(state => state.workouts);
+	const currentExercises = workouts.filter(workout => exerciseCard.name === workout.name);
+	const sortedCurrentExercises = currentExercises.sort(
+		(a, b) => new Date(b.date) - new Date(a.date)
+	);
+	const latestStats = sortedCurrentExercises.length
+		? sortedCurrentExercises[0]
+		: {sets: '0', reps: '0', weight: '0'};
 
-	function handleWeightInput(input) {
-		changeWeight(card.id, Number(input));
+	function selectDate(selectedDate) {
+		const date = selectedDate.toDateString();
+		addToPlanner(
+			date,
+			exerciseCard.name,
+			latestStats.sets,
+			latestStats.reps,
+			latestStats.weight
+		);
+		setIsCalendarVisible(false);
 	}
 
 	return (
 		<StyledCard>
 			<StyledCardHeader>
-				<StyledCardHeadline>{card.name}</StyledCardHeadline>
-				<BookmarkButton
-					onClick={() => {
-						debookmark(card.id);
-						setIsBookmarked(!isBookmarked);
-					}}
-				>
-					{isBookmarked ? (
-						<FilledBookmarkIcon id={card.id} />
+				<StyledCardHeadline>{exerciseCard.name}</StyledCardHeadline>
+				<BookmarkButton onClick={() => handleBookmark(exerciseCard.id)}>
+					{exerciseCard.isBookmarked ? (
+						<FilledBookmarkIcon id={exerciseCard.id} />
 					) : (
-						<NotFilledBookmarkIcon id={card.id} />
+						<NotFilledBookmarkIcon id={exerciseCard.id} />
 					)}
 				</BookmarkButton>
 				<StyledCalendarButton
 					onClick={() => {
-						setIsShown(!isShown);
+						setIsCalendarVisible(!isCalendarVisible);
 					}}
 				>
 					add to planner
 				</StyledCalendarButton>
-				{isShown && <IndexCalendar id={card.id} />}
+				{isCalendarVisible && <IndexCalendar onSelectDate={selectDate} />}
```

Comment:
I think you need to give your IndexCalendar the prop `exercise` as you have done in your `IndexCard.js`.
```suggestion
				{isCalendarVisible && <IndexCalendar exercise={exerciseCard} />
```

Reply:
I think here you need to add your add-button which calls the `addToPlanner` function like you did in the `IndexCard.js`:

```jsx
<button
		type="button"
		onClick={() => {
				addToPlanner(exerciseCard.id);
				setIsCalendarVisible(false);
					    }}>add
</button> 
```